### PR TITLE
DM-38514: Re-implement set-exposure-regions command

### PR DESCRIPTION
**Pull request by andy-slac**
The comand now uses new method of obscore manager which does approximately the same thing/ This simplifies command implementation.

### CODE REVIEW

File: python/lsst/dax/obscore/cli/cmd/commands.py [hunk 1]
```diff
@@ -30,10 +32,14 @@
     repo_argument,
     where_option,
 )
-from lsst.daf.butler.cli.utils import ButlerCommand, MWPath, split_commas
+from lsst.daf.butler.cli.utils import ButlerCommand, MWPath
 
 from ... import script
 
+split_commas = cast(
+    Callable[[click.Context, click.Option | click.Parameter, Any], Any], cmd_utils.split_commas
```
Comment (timj) [inline python/lsst/dax/obscore/cli/cmd/commands.py:40]: Remind me why we can't fix the annotation in butler?

File: python/lsst/dax/obscore/script/obscore_set_exposure_regions.py [hunk 2]
```diff
@@ -89,36 +87,23 @@ def obscore_set_exposure_regions(
...
-    # mess with the table contents directly. For that we need to have an access
-    # to the internals of the Registry and obscore manager.
-
     registry = butler.registry
-    assert isinstance(registry, SqlRegistry), "Registry must be SqlRegistry"
-
-    db = registry._db
-    obscore_mgr = registry._managers.obscore
-    if obscore_mgr is None:
+    if registry.obsCoreTableManager is None:
```
Comment (timj) [inline python/lsst/dax/obscore/script/obscore_set_exposure_regions.py:91]:
```suggestion
    if (obscore_mgr := registry.obsCoreTableManager) is None:
```

or maybe:

```python
obscore_mgr = registry.obsCoreTableManager
if obscore_mgr is None:
    raise ...
```
It seems odd to me that `registry.obsCoreTableManager` appears in two consecutive statements.

**Review by timj — APPROVED** [review]

File: python/lsst/dax/obscore/cli/cmd/commands.py (hunk 1 above)
Comment (andy-slac) [reply python/lsst/dax/obscore/cli/cmd/commands.py:40]: I don't remember now, but I can try to fix it.

Comment (andy-slac) [reply python/lsst/dax/obscore/cli/cmd/commands.py:40]: Looks like mypy 1.1.1 does not even need that cast (or maybe it's newer click). It complained about split_commas at USDF, but at my local machine with latest everything it looks fine.

Comment (timj) [reply python/lsst/dax/obscore/cli/cmd/commands.py:40]: USDF has types-click installed which breaks everything (we are trying to remove it from rubin-env-developer).

File: python/lsst/dax/obscore/script/obscore_set_exposure_regions.py (hunk 2 above)
Comment (andy-slac) [reply python/lsst/dax/obscore/script/obscore_set_exposure_regions.py:91]: I tried that, but for some reason mypy does not like that, it complains that `obscore_mgr` could be `None` inside `_count_missing` below. Not sure why, looks like mypy is confused by closures.

File: python/lsst/dax/obscore/cli/cmd/commands.py (hunk 1 above)
Comment (andy-slac) [reply python/lsst/dax/obscore/cli/cmd/commands.py:40]: So, if I remove those casts (there are also similar casts in daf_butler) people developing at USDF will suffer? Should I keep them for now?

Comment (timj) [reply python/lsst/dax/obscore/cli/cmd/commands.py:40]: I think that we are the main mypy users of daf_butler click tooling so I think it's fine to delete them (and other packages will break if types-click is present). @ktlim should be removing types-click in the next release.

Comment (andy-slac) [reply python/lsst/dax/obscore/cli/cmd/commands.py:40]: OK, I'll remove casts from dax_obscore and daf_butler.

**Pull request merged**